lib/Optimizer/Transforms/LoopUnroll.cpp

```diff
@@ -171,3 +171,21 @@ class LoopUnrollPass : public cudaq::opt::impl::LoopUnrollBase<LoopUnrollPass> {
   }
 };
 } // namespace
+
+void cudaq::opt::addUnrollingPipeline(OpPassManager &pm) {
+  pm.addPass(createCanonicalizerPass());
+  pm.addNestedPass<func::FuncOp>(createClassicalMemToReg());
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createLoopNormalize());
+  pm.addPass(createCanonicalizerPass());
+  constexpr LoopUnrollOptions luo{/*threshold=*/50, /*signalFailure=*/true};
```

Comment:
Should we add function arguments with defaults for these? I can envision someone wanting to add unrolling to the pipeline but does not want to signal failure.